### DISCUSSION TITLE
Meshery on RaspberryPi 🎯

### DISCUSSION THREAD

**Discussion by adithyaakrishna** [meshery/meshery#3531] [Ideas]
**Description:** Running Meshery on Raspberry Pi 💻 

**Supported Boards:**

- [ ] Raspberry Pi 4 - In Progress
- [ ] Raspberry Pi 3 - Testing | Demo Link [TBA]
- [ ] Raspberry Pi Zero - In Progress

**Prerequisites:** 

- [Raspberry Pi Setup](https://www.raspberrypi.org/documentation/setup/)
- [Raspberry Pi Installation](https://www.raspberrypi.org/documentation/installation/)
- [Raspberry Pi Configuration](https://www.raspberrypi.org/documentation/configuration/)
- [Raspberry Pi Usage](https://www.raspberrypi.org/documentation/usage/)

**Comment by adithyaakrishna** [comment]
@leecalcote I was able to run Meshery on a RPi running **Ubuntu**. Now I'm trying to run it in Raspberry Pi OS itself. Once this is done, we can try to setup a K8s Cluster on the board and then deploy Meshery using K8s.

P.S. Running K8s and Meshery both on the Raspberry Pi may cause huge delay or reduce the performance. But should try once to be sure

CC @navendu-pottekkat

**Reply by pottekkat** [reply]
Awesome 💯

**Comment by dhruv0000** [comment]
@adithyaakrishna. Are you accessing the k8s cluster in your PC (Not through SSH, but using kubectl). If so, how are you setting up the kubeconfig file in your PC to access the k8s cluster in Raspberry Pi.

**Reply by adithyaakrishna** [reply]
I had a RPi4 which had kubectl and docker installed on Ubuntu. But the board was borrowed from my friend a few days back then had to return it day before y'day. Now the board I have is RPi3 MB+ having docker installed.
For now my goal is to run it in docker itself, then if I get my hands on the other board again, then use K8s.
